src/ufixed.jl

```diff
@@ -98,26 +92,28 @@ abs(x::UFixed) = x
 # Functions
 trunc{T<:UFixed}(x::T) = T(div(reinterpret(x), rawone(T))*rawone(T),0)
 floor{T<:UFixed}(x::T) = trunc(x)
-for T in UF
-    f = nbitsfrac(T)
-    R = rawtype(T)
-    roundmask = convert(R, 1<<(f-1))
-    k = 8*sizeof(R)-f
-    ceilmask  = (typemax(R)<<k)>>k
-    @eval begin
-        round(x::$T) = (y = trunc(x); return convert(rawtype($T), reinterpret(x)-reinterpret(y))&$roundmask>0 ? $T(y+one($T)) : y)
-         ceil(x::$T) = (y = trunc(x); return convert(rawtype($T), reinterpret(x)-reinterpret(y))&$ceilmask >0 ? $T(y+one($T)) : y)
+@generated function round{T,f}(x::UFixed{T,f})
+    mask = convert(T, 1<<(f-1))
+    quote
+        y = trunc(x)
+        return convert(T, reinterpret(x)-reinterpret(y)) & $mask>0 ?
+                UFixed{T,f}(y+one(UFixed{T,f})) : y
+    end
+end
+@generated function ceil{T,f}(x::UFixed{T,f})
+    k = 8*sizeof(T)-f
+    mask = (typemax(T)<<k)>>k
+    quote
+        y = trunc(x)
+        return convert(T, reinterpret(x)-reinterpret(y)) & ($mask)>0 ?
+                UFixed{T,f}(y+one(UFixed{T,f})) : y
     end
 end
 
 trunc{T<:Integer}(::Type{T}, x::UFixed) = convert(T, div(reinterpret(x), rawone(x)))
 round{T<:Integer}(::Type{T}, x::UFixed) = round(T, reinterpret(x)/rawone(x))
 floor{T<:Integer}(::Type{T}, x::UFixed) = trunc(T, x)
  ceil{T<:Integer}(::Type{T}, x::UFixed) =  ceil(T, reinterpret(x)/rawone(x))
-trunc(x::UFixed) = trunc(Int, x)
```

Comment:
this is part of the second commit.  removing them fixes the bug that round(UFixed) returns an Int, not a UFixed.